src/prefect/new_task_engine.py

```diff
@@ -55,29 +50,12 @@
     collect_task_run_inputs,
     propose_state_sync,
 )
+from prefect.utilities.timeout import timeout, timeout_async
 
 P = ParamSpec("P")
 R = TypeVar("R")
 
 
-@asynccontextmanager
-async def timeout_async(seconds: Optional[float] = None):
-    try:
-        with AsyncCancelScope(timeout=seconds):
-            yield
-    except CancelledError:
-        raise TimeoutError(f"Task timed out after {seconds} second(s).")
```

Comment:
note that this was "superficially async" since the cancel scope itself is a sync context manager